Hackathon: Refactor a large number of the legacy stories to the new inline format.

benbowler opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #9408

## Relevant technical choices

I made great progress refactoring legacy stories from the `/stories/*` stories into the new format. I ran out of time so will pick up the following at a future hackathon:

1. Finish work by refactoring `generateReportBasedWidgetStories` module stories into their modules.
2. Find out how to use [react-router-dom in the new story format](https://github.com/google/site-kit-wp/blob/236d19ed491309c449468e91b8b22c4eb2a4d58c/assets/js/components/settings/SettingsModules.stories.js#L79-L83).
3. Move `tokens` and `type-scale` stories to a storybook Styles folder.
4. Refactor the more complex `widgets` stories to a logical place.
5. Complete and audit and validate 

*Update 17 December 2024:* I took the hackathon and fixed up the VRTs to make sure none are missing and the PR is ready for review.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil reviewed (changes requested):
Hi @benbowler, this looks like a great refactor on the whole, but it would be a lot easier to review if it adhered to these AC points:

- Stories should be logically organized together in a way which is consistent with how new stories are organized today
- Stories with related VRT scenarios should maintain the same target scenario files for now (i.e. no changes to VRT should happen as part of this refactoring)

Is there any reason this can't be achieved?

benbowler commented:
Hey @techanvil, the way I worked through the stories prioritised point 1:

> Stories should be logically organized together in a way which is consistent with how new stories are organized today

I moved stories inline with the relevant components, as well as often updating the folder/path of the stories such as moving `Accordion` from `Global` to `Components`. This causes a change in VRT image name. *Is there something I've missed that allows me to force a renamed/moved story to use reference images of a different name?*

---

One other element is that many legacy stories were a single story containing all variants such as [links](https://github.com/google/site-kit-wp/blob/769d6d4831144ec48afefd41d183c9f9478240b1/stories/links.stories.js). To bring this to match how we structure modern stories with separate stories for each variant I have split these but to keep a single VRT, I add [a single story with all variants for the VRT](https://github.com/google/site-kit-wp/blob/e8c70bd2f13fca3095de3633bc5b1670f17bffaa/assets/js/components/Link.stories.js#L221-L322). This doesn't directly effect above if I can link these VRTs to the existing reference image names.

techanvil commented on assets/js/components/layout/Layout.stories.js:
```diff
@@ -0,0 +1,63 @@
+ * Internal dependencies
+ */
+
+export default {
+	title: 'Global/Layout',
+	component: Layout,
+};
+
```
Please move the `export default` statement to the bottom of the file, all the other stories do follow this pattern from what I can see.

techanvil commented on assets/js/components/layout/Layout.stories.js:
```diff
@@ -0,0 +1,63 @@
+	);
+}
+
+export const Default = Template.bind( {} );
+Default.storyName = 'Default';
+Default.scenario = {};
+Default.args = {
```
It's not essential for every story to have one named `Default`. It would be more practical to name and configure this as follows to retain the current structure and VRT filename. This general principle should be applied to the rest of the stories in this PR where possible.

```suggestion
export const WithHeaderFooterAndCTAs = Template.bind( {} );
WithHeaderFooterAndCTAs.storyName = 'Layout with Header Footer and CTAs';
WithHeaderFooterAndCTAs.scenario = {
	label: 'Global/Layout/Layout with Header Footer and CTAs',
};
WithHeaderFooterAndCTAs.args = {
```

techanvil commented on assets/js/components/layout/Layout.stories.js:
```diff
@@ -0,0 +1,63 @@
+};
+
+export const WithHeaderAndFooter = Template.bind( {} );
+WithHeaderAndFooter.storyName = 'With Header and Footer';
+WithHeaderAndFooter.scenario = {};
```
As above:

```suggestion
WithHeaderAndFooter.storyName = 'Layout with Header and Footer';
WithHeaderAndFooter.scenario = {
	label: 'Global/Layout/With Header and Footer',
};
```

benbowler commented:
Hey @techanvil, great suggestions, I reverted the updated reference images then went through and matched each story so that there are now no changes to VRT images, with the exception of the Checkbox component which didn't have a story previously so I added one for consistency.

I kept a few logical Default named stories as they seemed consistent and clear, feel free to highlight any you think should have a more appropriate name.

techanvil commented on assets/js/components/ImageRadio.stories.js:
```diff
@@ -1,7 +1,7 @@
  * Radio Component Stories.
  *
- * Site Kit by Google, Copyright 2022 Google LLC
+ * Site Kit by Google, Copyright 2024 Google LLC
```
As this file has been moved, we don't really need to update the date in the header.

```suggestion
 * Site Kit by Google, Copyright 2022 Google LLC
```

techanvil commented on assets/js/components/ReportZero.stories.js:
```diff
@@ -1,7 +1,7 @@
  * ReportZero stories.
  *
- * Site Kit by Google, Copyright 2021 Google LLC
+ * Site Kit by Google, Copyright 2024 Google LLC
```
As above:

```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

techanvil commented on assets/js/components/TableOverflowContainer.stories.js:
```diff
@@ -1,7 +1,7 @@
  * TableOverflowContainer stories.
  *
- * Site Kit by Google, Copyright 2021 Google LLC
+ * Site Kit by Google, Copyright 2024 Google LLC
```
As above:

```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

techanvil commented on assets/js/googlesitekit/components-gm2/Button.stories.js:
```diff
@@ -1,7 +1,7 @@
  * Button Component Stories.
  *
- * Site Kit by Google, Copyright 2021 Google LLC
+ * Site Kit by Google, Copyright 2024 Google LLC
```
There's no need to update this date:

```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

techanvil commented on assets/js/googlesitekit/components-gm2/Button.stories.js:
```diff
@@ -171,8 +171,10 @@ export function VRTStory() {
+VRTStory.storyName = 'All Buttons VRT';
 VRTStory.scenario = {
 	label: 'Global/Buttons',
+	delay: 3000,
```
Why was the delay added here?

benbowler replied:
I don't think it's needed, removing it.

techanvil commented on assets/js/googlesitekit/components-gm2/Menu.stories.js:
```diff
@@ -1,7 +1,7 @@
  * Menu Component Stories.
  *
- * Site Kit by Google, Copyright 2021 Google LLC
+ * Site Kit by Google, Copyright 2024 Google LLC
```
As above:

```suggestion
 * Site Kit by Google, Copyright 2021 Google LLC
```

techanvil reviewed (changes requested):
Thanks @benbowler! I've reviewed all of the changes and checked the stories in Storybook. It's generally looking great, I have left a number of comments for your attention, most of which are pretty minor.

Also noting that this wouldn't be practical for a member of the QA team to test, so I've been doing QA:Eng testing as part of the CR process. Once you've addressed the review comments, I can re-review the affected stories and wrap up the QA as well.

benbowler commented on assets/js/components/Link.stories.js:
```diff
@@ -0,0 +1,328 @@
+			</p>
+			<p>
+				<SecondaryLinkButtonWithIconPrefixHovered
+					{ ...SecondaryLinkButtonWithIconPrefixHovered.args }
+				/>
+			</p>
+			<p>
+				<SmallLink { ...SmallLink.args } />
+			</p>
+			<p>
+				<InverseLink { ...InverseLink.args } />
+			</p>
+			<p>
+				<BackLink { ...BackLink.args } />
+			</p>
+			<p>
+				<ExternalLink { ...ExternalLink.args } />
+			</p>
+			<p>
+				<ExternalLinkWithVisuallyHiddenContent
+					{ ...ExternalLinkWithVisuallyHiddenContent.args }
+				/>
+			</p>
+			<p>
+				<AllCapsLink { ...AllCapsLink.args } />
+			</p>
+			<p>
+				<AllCapsLinkWithArrow { ...AllCapsLinkWithArrow.args } />
+			</p>
+			<p>
+				<InverseAllCapsLinkWithArrow
+					{ ...InverseAllCapsLinkWithArrow.args }
+				/>
+			</p>
+			<p>
+				<DangerLink { ...DangerLink.args } />
+			</p>
+			<p>
+				<DisabledLink { ...DisabledLink.args } />
+			</p>
```
I applied this to other stories such as Checkbox and Button.

benbowler commented:
Thanks for taking the time to look at this big PR. I have addressed each of the points mentioned.

techanvil commented on assets/js/components/settings/SettingsApp.stories.js:
```diff
@@ -16,38 +16,20 @@
 
 function Template() {
 	return (
 		<Layout>
```
We shouldn't wrap these components in `Layout`, as the result is not representative of how the plugin looks. I'll apply this change myself to save a round trip.

```suggestion
```

_Story:_
![image](https://github.com/user-attachments/assets/33772976-d16a-4f42-adcc-69014cd9deaf)


_Plugin:_
![image](https://github.com/user-attachments/assets/d02e37fd-5a99-422d-a9e5-f2b9c0f17335)

techanvil reviewed (approved):
LGTM, nice one @benbowler! Great work.

I've confirmed the failing E2E tests are unrelated to the changes in this PR, as is the one failing VRT scenario.

:white_check_mark:
